### PR TITLE
Support multiple weekdays and ordinal days

### PR DESCRIPTION
**Pull request by jbondia**
Allowing any weekday, month, ordinal days comma separated values.

### CODE REVIEW

**Comment by adamchainz** [comment]
1. Please add tests! I see you have a TODO comment for that in your second commit message
2. Remove all `pylint` comments - we don't lint this repository with pylint, if you do personally it's your responsibility to disable it whilst editing

**Comment by jbondia** [comment]
1. Yep, sorry about that.
2. Understood

I will do it after work.

Thanks

**Comment by adamchainz** [comment]
Also:
3. Add yourself to AUTHORS.rst as a contributor! Alphabetical order by first name please.
4. Add a line to HISTORY.rst underneath 'New release notes here'.

**Comment by jbondia** [comment]
If I am not wrong:
1. Deleted pylint comments.
2. Added tests for weekdays and ordinal days with comma separated values (months already supported).
3. Added as contributor with first name alphabetical order.
4. Added history line describing the changes.

Regards,

File: HISTORY.rst [hunk 1]
```diff
@@ -8,6 +8,11 @@ Pending Release
 
 * New release notes here
 
+1.0.2 (2016-04-27)
```
Comment (adamchainz) [inline HISTORY.rst]: please don't add release header, i'll do that on release

File: pretty_cron/api.py [hunk 2]
```diff
@@ -65,13 +72,18 @@ def _pretty_date(month_day, month, week_day):
             )
 
         pretty_date = " and ".join(
+            # pylint: disable=W0141
```
Comment (adamchainz) [inline pretty_cron/api.py]: pylint comment still here

File: pretty_cron/api.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
 def prettify_cron(expression):
+    """
+    Returns human readable cron-tab configuration
+    """
     pieces = []
-    for i, piece in enumerate(expression.split(" ")):
-        if i == 3 and ',' in piece:  # support comma-separated values for month
+    for piece in expression.split(" "):
+        # support comma-separated values for ordinal days, weekdays and months
```
Comment (adamchainz) [inline pretty_cron/api.py:11]: if this only supports days, weekdays, and months, shouldn't the conditional include, like before, `if i in (1, 2, 3)` ? otherwise you're allowing `,` in hours

File: pretty_cron/api.py [hunk 4]
```diff
@@ -65,13 +72,18 @@ def _pretty_date(month_day, month, week_day):
...
             filter(None, (month_day_date, week_day_date))
+            # pylint: enable=W0141
         )
 
     return pretty_date
 
 
 def _human_month(month):
+    """
+    Cron-tab to month
```
Comment (adamchainz) [inline pretty_cron/api.py]: vacant docstring, please remove (it says no more than the function name)

File: pretty_cron/api.py [hunk 5]
```diff
@@ -106,7 +121,17 @@ def _human_list(listy):
 
 
 def _human_week_day(day):
-    return _WEEKDAYS[day]
+    """
+    Cron-tab to weekday
```
Comment (adamchainz) [inline pretty_cron/api.py]: ditto, vacant docstring, please remove (just because pylint tells you to do something...)

File: pretty_cron/api.py [hunk 6]
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
 
 
 def _ordinal(n):
-    if 10 <= (n % 100) < 20:
-        suffix = 'th'
+    """
+    Cron-tab to numeric day
```
Comment (adamchainz) [inline pretty_cron/api.py]: ditto plz remove

File: pretty_cron/api.py [hunk 7]
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
...
+            suffix = _ORDINAL_SUFFIXES.get(d % 10, 'th')
+        ordinal_days.append(str(d) + suffix)
 
-    return str(n) + suffix
+    return _human_list(ordinal_days)
 
 
 def _pretty_time(minute, hour):
+    """
+    Cron-tab to time
```
Comment (adamchainz) [inline pretty_cron/api.py]: ditto

File: pretty_cron/api.py [hunk 8]
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
 
 
 def _ordinal(n):
-    if 10 <= (n % 100) < 20:
-        suffix = 'th'
+    """
+    Cron-tab to numeric day
+    """
+    if isinstance(n, tuple):
```
Comment (adamchainz) [inline pretty_cron/api.py]: this is simpler as

``` py
if not isinstance(n, tuple):
    n = (n,)
```

File: pretty_cron/api.py (hunk 3 above)
Comment (jbondia) [inline pretty_cron/api.py]: I guess you are right. Sorry about this.
What about 'i != 0 and [...]' instead?

File: pretty_cron/api.py (hunk 5 above)
Comment (jbondia) [inline pretty_cron/api.py]: As before, you are right. Do you prefer me to remove all the docstrings or write them in the right way explaining parameters and return stuff?

File: pretty_cron/api.py (hunk 8 above)
Comment (jbondia) [inline pretty_cron/api.py]: OMG, yeah. Thanks. I guess we also could apply this change at _human_month and _human_week_day?

File: .gitignore [hunk 9]
```diff
@@ -3,6 +3,10 @@
 # C extensions
 *.so
 
+# vim temporal files
```
Comment (adamchainz) [inline .gitignore]: please don't do this and introduce unnecessary change, if your editor generates extra temporary files then ignore them with a global git ignore like https://help.github.com/articles/ignoring-files/

Comment (jbondia) [inline .gitignore]: Sorry Adam, and thank you for your patience. I am learning quite a lot of things today..

Comment (adamchainz) [inline .gitignore]: No worries, glad to help! Thanks for working with me in my push for perfection 👍

**Comment by adamchainz** [comment]
Merged in edd42a5df913b97649b72cacc1e5a29505a04b02 , thanks!

**Comment by jbondia** [comment]
Thanks a lot for your help Adam

**Comment by adamchainz** [comment]
Sorry, it was 0e5d729e0c0d50715514de502be669ed9bdb2267 that merged it, commit SHA got changed when I rebased it onto master

**Comment by adamchainz** [comment]
If you were waiting for PyPI release, I just did it as 1.0.2: https://pypi.python.org/pypi/pretty-cron

**Comment by jbondia** [comment]
Thanks a lot, Adam!

Julio Bondia Barceló
